### DISCUSSION TITLE
Dynamite Core Team Meeting 19th April

### DISCUSSION THREAD

**Discussion by boeckera** [dynamics-of-stellar-systems/dynamite#380] [Announcements]
Hola @dynamics-of-stellar-systems/dynamite-maintainers,

at this week's meeting at **1pm (onsite: Glenn's office, online: dynamite zoom link)** we will mainly talk about the issue of storage space on VSC and possible deletion of some DYNAMITE files that are currently saved automatically.

Please, if you have any other topics you want to bring up respond below.
To indicate your participation you can react with an appropriate emoji of your choice.

Looking forward to seeing you all!

Un saludo,
Alina

**Comment by maindlt** [comment]
Hi there, does anyone have time to test PR #365 (about labels in the chi2 plot, would be a quick win 😉) and/or have quick look at PR #379 (about citing Decomposition in the docs)...? Cheers and see you Friday, Thomas

**Reply by boeckera** [reply]
Hi Thomas,

could you kindly describe a quick instruction again how to test this? I installed dynamite on my MAC, but now what do I do haha
We can also go through it together on Friday :)

Thank you!

**Reply by maindlt** [reply]
Hi Alina,

many thanks for taking the time to test! :-)

PR #365 is all about the $\chi^2$ plot which is stored in `plots/kinchi2_plot.png` in the output directory. More specifically, the PR fixes the labels of the x and y axes and the colorbar and also makes sure the colorbar does not overlap with the actual plot.

We can of course do this together on Friday, but if you are interested please have a look at my ideas for testing below ;-)

Recommended testing:

- Check out the pull request's branch `chi2_plot_fixes`. If DYNAMITE was installed already, the legacy Fortran does not need to be recompiled for this PR, but a `python setup.py install --user` is still necessary to update all the Python files.
- Let's use the test script `test_nnls.py` in the `dev_tests/` folder for testing. This script reads the config file `user_test_config_ml.yaml` in the same directory.
- Running the script will create an empty $\chi^2$ plot because all parameters but one (`ml`) are fixed. To create a minimal $\chi^2$ plot, one additional parameter needs to be set to `fixed: False` in the configuration file. Before, this created a plot where the colorbar overlapped the x-axis. This should not be the case any more here. I used `fixed: False` for the black hole `m` parameter, perhaps you can use another one, run the script with this new configuration,  and verify that the colorbar no longer overlaps and has a meaningful caption.
- Finally, let's create a more complex plot and verify that all the captions make sense. For this, I increased the maximum number of models to `n_max_mods : 6` in the `parameter_space_settings` of the config file and set a few more parameters to `fixed: False` (I chose the dark halo `c` and the stars `q`, but any parameter that has some entries in its `par_generator_settings:` qualifies). Running the script again takes a bit longer (about 2.25 min on my 2020 Mac) and should result in a more complex `plots/kinchi2_plot.png` with meaningful labels on the axes and the colorbar.
- Done :-)

Cheers,
Thomas
